ocs_ci/utility/ibmcloud.py

```diff
@@ -520,6 +522,280 @@ def delete_volume_id(self, volume):
+        """
+        Start the nodes on IBM Cloud
+        Args:
+            nodes (list): The OCS objects of the nodes
+        """
+        # logger.info(nodes)
+        for node in nodes:
+            # logger.info(node.get())
+            cmd = f"ibmcloud is instance-start {node.name}"
+            out = run_ibmcloud_cmd(cmd)
+            logger.info(f"Node start command output: {out}")
+
+    def stop_nodes(self, nodes, force=True, wait=True):
+        """
+        Stop the nodes on IBM Cloud
+        Args:
+            nodes (list): The OCS objects of the nodes
+            force (bool): True for VM ungraceful power off, False for graceful VM shutdown
+            wait (bool): Wait for the VMs to stop
+        """
+        for node in nodes:
+            cmd = f"ibmcloud is instance-stop {node.name} --force={force}"
+            out = run_ibmcloud_cmd(cmd)
+            logger.info(f"Node Stop command output: {out}")
+
+        if wait:
+            for node in nodes:
+                sample = TimeoutSampler(
+                    timeout=300,
+                    sleep=10,
+                    func=self.check_node_status,
+                    node_name=node.name,
+                    node_status=constants.STATUS_STOPPED,
+                )
+                sample.wait_for_func_status(result=True)
+
+    def restart_nodes_by_stop_and_start(
+        self, nodes, wait=True, force=True, timeout=300
+    ):
+        """
+        Restart nodes by stopping and starting VM in IBM Cloud
+        Args:
+            nodes (list): The OCS objects of the nodes
+            wait (bool): True in case wait for status is needed, False otherwise
+            force (bool): True for force instance stop, False otherwise
+            timeout (int): Timeout for the command, defaults to 300 seconds.
+        """
+        logger.info(f"Stopping instances {list(nodes)}")
+        self.stop_nodes(nodes=nodes, force=force)
+        if wait:
+            for node in nodes:
+                sample = TimeoutSampler(
+                    timeout=timeout,
+                    sleep=10,
+                    func=self.check_node_status,
+                    node_name=node.name,
+                    node_status=constants.STATUS_STOPPED,
+                )
+                sample.wait_for_func_status(result=True)
+        logger.info(f"Starting instances {list(nodes)}")
+
+        self.start_nodes(nodes=nodes)
+        if wait:
+            for node in nodes:
+                sample = TimeoutSampler(
+                    timeout=timeout,
+                    sleep=10,
+                    func=self.check_node_status,
+                    node_name=node.name,
+                    node_status=constants.STATUS_RUNNING.lower(),
+                )
+                sample.wait_for_func_status(result=True)
+
+    def check_node_status(self, node_name, node_status):
+        """
+        Check the node status in IBM cloud
+
+        Args:
+            node_name (str): Node name
+
```

Comment:
Please add the new param `node_status` in the docstring.

Reply:
I have Updated the PR